improver/utilities/mathematical_operations.py

```diff
@@ -200,68 +230,64 @@ def perform_integration(
             lower_bounds_cube (iris.cube.Cube):
                 Cube containing the lower bounds to be used during the
                 integration.
-            integrated_cube (iris.cube.Cube):
-                Cube that will be used for storing the output of the
-                integration containing the most appropriate coordinates.
 
         Returns:
             iris.cube.Cube:
                 Cube containing the output from the integration.
 
         """
-        # Create a zip for looping over.
-        levels_tuple = list(zip(
+        def skip_slice(upper_bound, lower_bound, direction,
+                       start_point, end_point):
+            """Conditions under which a slice should not be included in
+            the integrated total"""
+            if start_point:
+                if direction == "positive" and lower_bound < start_point:
+                    return True
+                if direction == "negative" and upper_bound > start_point:
+                    return True
+            if end_point:
+                if direction == "positive" and upper_bound > end_point:
+                    return True
+                if direction == "negative" and lower_bound < end_point:
+                    return True
+            return False
+
+        data = []
+        coord_points = []
+        coord_bounds = []
+        stride_sum = 0
+        template = iris.cube.CubeList([])
+        levels_tuple = zip(
             upper_bounds_cube.slices_over(self.coord_name_to_integrate),
-            lower_bounds_cube.slices_over(self.coord_name_to_integrate),
-            integrated_cube.slices_over(self.coord_name_to_integrate)))
+            lower_bounds_cube.slices_over(self.coord_name_to_integrate))
+
+        for (upper_bounds_slice, lower_bounds_slice) in (levels_tuple):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary parens after 'in' keyword](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=4810470)